packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx

```diff
@@ -31,7 +31,10 @@ export const CloudPulseDashboardSelect = React.memo(
 
     const serviceTypes: string[] = formattedServiceTypes(serviceTypesList);
     const serviceTypeMap: Map<string, string> = new Map(
-      serviceTypesList?.data.map((item) => [item.service_type, item.label])
+      (serviceTypesList?.data || []).map((item) => [
+        item?.service_type ?? '',
+        item?.label ?? '',
+      ])
```

Comment:
```suggestion
      (serviceTypesList?.data || []).filter((item) => item?.serviceType !== undefined).map((item) => [
        item?.service_type ?? '',
        item?.label ?? '',
      ])
```

its better to filter out undefined service types